### PR TITLE
More C++ pixi commands, enable Wunused-parameter

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* some recycling from aborted #3921

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3922) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/3922)
- [Docs preview](https://rerun.io/preview/d1a9b18194697601b0590029541d448c5e00874c/docs) 
- [Examples preview](https://rerun.io/preview/d1a9b18194697601b0590029541d448c5e00874c/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: pixi.toml [hunk 1]
```diff
@@ -33,10 +33,16 @@ py-test = { cmd = "python -m pytest -vv rerun_py/tests/unit", depends_on = [
   "py-build",
 ] }
 
-cpp-prepare = "cmake -G Ninja -B build -S . -DCMAKE_BUILD_TYPE=Debug"
+cpp-prepare = "cmake -G Ninja -B build -S . -DCMAKE_BUILD_TYPE=Debug -DCMAKE_COMPILE_WARNING_AS_ERROR=ON"
```
Comment (jleibs) [inline pixi.toml:36]: I'm curious how much running this every time slows down the dev-cycle. Should we consider making this a no-op if the `build` folder already exists?

File: pixi.toml [hunk 2]
```diff
@@ -33,10 +33,16 @@ py-test = { cmd = "python -m pytest -vv rerun_py/tests/unit", depends_on = [
...
 cpp-build = { cmd = "cmake --build build --config Debug --target rerun_sdk_tests", depends_on = [
```
Comment (jleibs) [inline pixi.toml:37]: Should this be `cpp-build-tests` for consistency?

**Review by jleibs — APPROVED** [review]

File: pixi.toml (hunk 2 above)
Comment (Wumpf) [reply pixi.toml:37]: yes!

File: pixi.toml (hunk 1 above)
Comment (Wumpf) [reply pixi.toml:36]: Hmm ideally I'd like to parameterize this to make it opt-in. CI should always opt-in

Comment (emilk) [reply pixi.toml:36]: I’ve documented how to do that in the pixi.toml file

Comment (Wumpf) [reply pixi.toml:36]: thanks, and that's nice and easy. But there doesn't seem to be a way to pass things through to the `depends_on` part? I'd like to be able to `cpp-build -DCMAKE_COMPILE_WARNING_AS_ERROR=ON` which would forward this argument to the underlying `cpp-prepare` 🤔

Comment (emilk) [reply pixi.toml:36]: Ask on the pixi Discord :)

**Comment by teh-cmc** [comment]
We still have those `build_all.sh` scripts around. Can they go away now?

**Comment by Wumpf** [comment]
Some of them are still used by Ci which doesn't have pixi yet (there's a separate issue about this). After that yes please

File: pixi.toml (hunk 1 above)
Comment (Wumpf) [reply pixi.toml]: env vars is the way to go for now it seems!

**Pull request merged**